src/analysis/mesh/qgsmeshcontours.h

```diff
@@ -0,0 +1,97 @@
+/***************************************************************************
+                          qgsmeshcontours.h
+                          -----------------
+    begin                : September 25th, 2019
+    copyright            : (C) 2018 by Peter Petrik
+    email                : zilolv at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMESHCONTOURS_H
+#define QGSMESHCONTOURS_H
+
+#include <QMap>
+#include <QString>
+#include <QStringList>
+#include <QVector>
+#include <QPair>
+#include <QList>
+#include <memory>
+#include <limits>
+
+#include "qgsmeshdataprovider.h"
+#include "qgsfeature.h"
+#include "qgis_analysis.h"
+#include "qgis_sip.h"
+#include "qgstriangularmesh.h"
+#include "qgsmeshrenderersettings.h"
+
+class QgsMeshLayer;
+class QgsPoint;
+class QgsGeometry;
+
+/**
+ * \ingroup analysis
+ * \class QgsMeshContours
+ *
+ * Exporter of countour lines or polygons from a mesh layer.
+ *
+ * \since QGIS 3.12
+ */
+class ANALYSIS_EXPORT QgsMeshContours
+{
+  public:
+
+    /**
+     * Contructs the mesh contours exporter.
+     * Caches the native and triangular mesh from data provider
+     * \param layer mesh layer to be associated with this exporter
+     */
+    QgsMeshContours( QgsMeshLayer *layer );
+    ~QgsMeshContours();
+
+    /**
+     * Exports multi line string containing the contour line for particular dataset and value
+     * \param index dataset index
+     * \param value value of the contour line
+     * \param method for datasets defined on faces, the method will be used to convert data to vertices
+     * \returns MultiLineString geometry containing contour lines
+     */
+    QgsGeometry exportLines( const QgsMeshDatasetIndex &index,
```

Comment:
Would be good to add qgsfeedback arguments here to allow early cancellation

Reply:
done